src/cpp/core/include/core/SafeConvert.hpp

```diff
@@ -137,10 +137,18 @@ boost::optional<TOutput> numberTo(TInput input)
    return result;
 }
 
+// Indexing into an empty vector causes assertion failures on some platforms.
+// Cannot rely on vector::data() as it is not guaranteed to be nullptr for
+// empty vectors.
+template <typename T>
+T* safe_vec_addr(std::vector<T>& vec)
```

Comment:
I don't feel like this really belongs in a utility method in `safe_convert` myself, but @jmcphers probably knows better. I'd be tempted to say we should create our own wrapper container for `std::vector` and require us to use that instead of the regular vector (perhaps via compiler flags to spit out deprecation warnings or errors?)

Reply:
So thinking wrapping vector in all cases might be a bigger change than necessary vs. just a selective tweak. Still, that would require extra thinking versus a gigantic search and replace.

Reply:
I agree completely, but the downside with this approach is we have to remember this pattern everywhere going forward. But this small issue is likely not worth my suggestion, like you suggest.

Reply:
If I had to suggest where else we might move such a method, I'd say we'd put it in a collection utils namespace.